src/simplify_algebra.cpp

```diff
@@ -714,6 +716,131 @@ struct find_rsqrt
     }
 };
 
+template <class Op>
+static bool check_dims(const std::vector<instruction_ref>& vec_ins)
```

Comment:
This should probably be called `same_ops` since it can check more than just dims.